### PR TITLE
8334295: CTW: update modules

### PR DESCRIPTION
**Pull request by lepestock**
This updates Compile-the-world modules, adding some new ones.
Testing: added tests + tier1 on 5 supported platforms, with and without debug enabled.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8334295](https://bugs.openjdk.org/browse/JDK-8334295): CTW: update modules (**Enhancement** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19738/head:pull/19738` \
`$ git checkout pull/19738`

Update a local copy of the PR: \
`$ git checkout pull/19738` \
`$ git pull https://git.openjdk.org/jdk.git pull/19738/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19738`

View PR using the GUI difftool: \
`$ git pr show -t 19738`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19738.diff">https://git.openjdk.org/jdk/pull/19738.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19738#issuecomment-2179846746)

### CODE REVIEW

File: test/hotspot/jtreg/applications/ctw/modules/generate.bash [hunk 1]
```diff
@@ -30,7 +30,7 @@ do
     echo creating $file for $module...
     cat > $file <<EOF
 /*
- * Copyright (c) 2017, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2017, 2024, Oracle and/or its affiliates. All rights reserved.
```
Comment (shipilev) [inline test/hotspot/jtreg/applications/ctw/modules/generate.bash:33]: Hm. So if the date had changed here, why it did not change in the existing files?

**Review by shipilev — COMMENTED** [review]
Looks okay, but look at the warnings: the PR should probably be version `24`, not `23`; and PR should probably have a `/label` so that it shows up on mailing lists.

File: test/hotspot/jtreg/applications/ctw/modules/generate.bash (hunk 1 above)
Comment (lepestock) [reply test/hotspot/jtreg/applications/ctw/modules/generate.bash:33]: 1. I generated only the files for the new modules, not changing the existing ones (as they would not change);
2. I've manually edited the generated headers, removing the "2017" part. I'm not a layer, but it seems reasonable as the new files' code didn't exist before 2024, despite it being a standard and template one.

Comment (shipilev) [reply test/hotspot/jtreg/applications/ctw/modules/generate.bash:33]: IIRC, there is already a precedent for regenerating and updating all the files when template file change, e.g. in ByteBuffers / VarHandles. The problem with manual intervention is that a next person that regenerates the files from the template would expect no changes in files when template did not change.

Comment (lepestock) [reply test/hotspot/jtreg/applications/ctw/modules/generate.bash:33]: Ok, let me suggest a compromise. I modified the generator to have 2 modes - to generate a new file and to update an existing one.
Other files have been edited manually anyways (some are split in 2 parts, some have got manually added `@requires` blocks, etc.), which means that there will be no clear generation. But at least the generator became more useful and concise.

**Comment by TobiHartmann** [comment]
I think this should go to `hotspot-compiler` not `compiler` which is for javac.

**Review by shipilev — COMMENTED** [review]
Apologies, but this is worse: now the generate script would not overwrite the files, which would be the expected behavior when template changes? Let's rollback to https://github.com/openjdk/jdk/pull/19738/commits/a4605c9a977ede23ce86de1df8267a176bf40fa9, which is less problematic. I'll approve then.

**Comment by lepestock** [comment]
> Let's rollback to https://github.com/openjdk/jdk/commit/a4605c9a977ede23ce86de1df8267a176bf40fa9, which is less problematic. I'll approve then.

Ok. I've rolled back the generator, in the c4dff565f4256f1864b37cc7e1cfe467f9c2311e. Thanks in advance.

**Review by shipilev — APPROVED** [review]

**Comment by shipilev** [comment]
Not sure if this qualifies as trivial, so I'll wait for 2nd review before sponsoring :)

**Comment by vnkozlov** [comment]
> Testing: added tests + tier1 on 5 supported platforms, with and without debug enabled.
@lepestock you need to run all first 4 tiers in mach5 for testing CTW.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me too. I can sponsor once your testing passed.

**Comment by lepestock** [comment]
Testing tiers 1-4 passed with no failures spotted.

**Comment by vnkozlov** [comment]
> Testing tiers 1-4 passed with no failures spotted.

Thank you for testing.

**Pull request closed without merging**